### PR TITLE
WIP: Modify ProGuard rules to automatically shrink object files

### PR DESCRIPTION
**Pull request by hagau**
## Description, Motivation and Context
Allow ProGuard to automatically shrink object files.

## How Has This Been Tested?
- Testsuite passes.
- Key generation, key use, key update, key backup have been manually tested.
- SSH authentication api has been tested.
- openpgp-api works with K9.
- Manual inspection reveals android.support.v7.view.menu.** to be mostly obfuscated; there are 3 classes not obfuscated. I don't have a Samsung device to test this.

### CODE REVIEW

File: OpenKeychain/build.gradle [hunk 1]
```diff
@@ -207,10 +213,17 @@ android {
...
 
         debug {
-            // disable ProGuard for faster compile times for debug builds
-            minifyEnabled false
+            minifyEnabled true
+
+            proguardFiles = buildTypes.release.proguardFiles
 
-            multiDexEnabled true
+            testProguardFiles getDefaultProguardFile('proguard-android.txt'),
```
Comment (dschuermann) [inline OpenKeychain/build.gradle]: ``testProguardFiles`` only needs the additional files, i.e., ``'proguard-rules-test.pro'``

https://github.com/googlesamples/android-testing-templates/tree/master/AndroidTestingBlueprint#proguard

Comment (dschuermann) [reply OpenKeychain/build.gradle]: I did this here: https://github.com/open-keychain/open-keychain/tree/hagau-proguard

Comment (hagau) [reply OpenKeychain/build.gradle]: Thanks, I'm sure I have read that page at some point, but forgot to clean up that part of the patch.

**Comment by dschuermann** [comment]
I tested this and looks like everything is working for me.

In comparison with enabling multidex, it is even faster to build in my tests (I hope I did it right). Looks like the dexing process takes really much longer for more methods/classes.

``./gradlew assembleFdroidDebug`` with proguard enabled: 
Total time: 1 mins 13.888 secs

``./gradlew assembleFdroidDebug`` with proguard disabled but multidex enabled: 
Total time: 1 mins 22.882 secs

(I did a ``./gradlew clean`` before each run)

@Valodim Any comments?

**Comment by hagau** [comment]
Using [dexcount-gradle-plugin](https://github.com/KeepSafe/dexcount-gradle-plugin) I collected some numbers:

`Total methods in OpenKeychain-fdroid-debug.apk: 43014 (65.64% used)`
`Total fields in OpenKeychain-fdroid-debug.apk:  35817 (54.65% used)`
`Total classes in OpenKeychain-fdroid-debug.apk:  6476 (9.88% used)`

(`(x% used)` means `x%` of 2^16)

the largest contributors to method counts:
`11730 android.support`
`9832 org.bouncycastle`
`9669 org.sufficientlysecure`

So even with these rather crude rules there is a lot of room left now.

**Comment by dschuermann** [comment]
Merged, but only enabled for release builds!

**Pull request merged**